### PR TITLE
Scaffold for the masses

### PR DESCRIPTION
**Pull request by nofxx**
This makes dead simple to use the scaffold on other arduinos and programmers.
Tested with duemilanove, usbasp and a bunch of chips.
Adds some README too.

Thank you for this project.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -22,7 +22,7 @@ void configure(void)
   TCCR0B = _BV(CS00) | _BV(CS01);
 
   /* enable the TIMER0 overflow interrupt */
-  TIMSK0 = _BV(TOIE0);
+  TIMSK0 = _BV(TOIE0); // TIMSK in ATTiny
```
Comment (sw17ch) [inline src/main.c:25]: Should there be more code in this file to support the ATTiny? As it stands, I'm probably going to toss the changes in main.c. I don't really see the intent here.

Comment (nofxx) [inline src/main.c:25]: For one, it shows how easy it is to target tiny ;)
But, the error on attiny is a bit cryptic too, It took me some minutes to figure out...

File: Rakefile [hunk 2]
```diff
@@ -1,17 +1,28 @@
...
+PROG = 'scaffold'
+
+# AVR device
+MCU = ENV['MCU'] || 'atmega328p' # 'attiny85'
 PARTNO = ENV['PARTNO'] || MCU
+
+# Flash burner
+PROGRAMMER = ENV['PROGRAMMER'] || 'arduino' # usbasp
+SERIAL_PORT = ENV['SERIAL_PORT'] ||  Dir.glob(
+      "/dev/tty#{RUBY_PLATFORM =~ /linux/ ? 'USB' : '.usbmodel'}").first
```
Comment (nofxx) [inline Rakefile:14]: Just saw an error here... there should be an asterisk '*' on string ending here:

```
 '.usbmodel'}").first
```

To:

```
 '.usbmodel'}*").first
```

Please fix on merge, sorry.

**Comment by nofxx** [comment]
Ended up with my collection: https://github.com/nofxx/avr_scaffold

**Pull request closed without merging**